src/v/cluster/health_monitor_backend.cc

```diff
@@ -773,6 +779,11 @@ health_monitor_backend::get_cluster_health_overview(
         ret.unhealthy_reasons.emplace_back("nodes_down");
     }
 
+    // cluster is not healthy if some node have degraded disk
+    if (!ret.data_disk_degraded.empty()) {
+        ret.unhealthy_reasons.emplace_back("data_disk_degraded");
```

Comment:
I wonder if we should include it here as it will preclude rolling restarts.

Reply:
@ztlpn ah. I wasn't aware about how other components interact with this. Thought it is only for monitoring/alerting. 

I'm open to suggestions for how to best surface problems like this to customers in a generic way rather than them having to go and look at error logs or be aware that particular metrics are more important than others.

This is more of a nice to have to me rather than a customer/product request.

Reply:
> This is more of a nice to have to me rather than a customer/product request.

Makes sense, I think including it in the health report is nice, just don't mark the cluster unhealthy if it is not empty :) (looks weird, as having degraded disks is clearly not healthy, but we have to think about how it will be interpreted by the operator. There are some improvements planned there: https://redpandadata.atlassian.net/browse/ENG-227)